metabase/metabase
# Remove styled-system

Removes `styled-system` and its usage in order to help unblock #31258. Styled-system won't play nicely with the `MantineThemeProvider` given some conflicts in how the Mantine theme is structured and what `styled-system` expects out of certain theme context properties.

## The good
- Helps unblock #31258
- Removes yet another styling mechanism & dependency

## The bad:
- Styled system props were included on some of our most work-horse-y components, like `Link` and `Button`
This makes verification a bit of pain. Now happily I think we haven't used space and color props widely in a couple years so it's not as bad as it could be. Also the class of bug this would introduce is 

Here's what I've done to mitigate any weirdness:

- [x] Happily `yarn type-check` caught a certain class of usage and I've gone in and replaced any use of `mx, mt` etc with our `className` utility equivalent where I can. I realize this is a bit silly given that we're about to adopt a UI library that uses some of this same methodology, but I think until that time I'd rather not convert to using `metabase/styled-components/theme/space` or otherwise try and patch things. Any introduction of new components like Button and Link that are Mantine based are going to require prop cleanup at their individual points of usage anyway.
- [x] In addition to the type check, I've manually searched for props from `styled-system` in the code and addressed their usage in the same manner as above.
- [x] space
- [x] hover
- [x] fontSize
- [x] color - the text component handles this ok, need to check `Icon` mostly from the looks of it.

## How to verify:
Sadly verification on this one might be tricky and a bit manual. But if you open up Metabase and click around you shouldn't see any spacing irregularities

I'm a little unsure if we can use Percy etc here to help with any visual regressions. I tried following the doc for that but it only seemed to test visualizations if I was reading the output correctly.
